core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -30,3 +30,13 @@ class SparkException(message: String, cause: Throwable)
  */
 private[spark] class SparkDriverExecutionException(cause: Throwable)
   extends SparkException("Execution error", cause)
+
+/**
+ * Exception indicating an error internal to Spark -- it is in an inconsistent state, not due
+ * to any error by the user
+ */
+class SparkIllegalStateException(message: String, cause: Throwable)
```

Comment:
I was hoping to create an exception which would make it clear that the user is totally free from blame.  I find IllegalStateExceptions can be confusing -- did this condition result because the user violated some precondition, misused the api etc?  Or is it a spark bug?

I felt this would be more clear, but its not critical.